### ISSUE TITLE
[VSCode] Binary file automatic reloading

### ISSUE THREAD

**Issue by akenejie** [TypeFox/open-collaboration-tools#61]
Text files are reflected to the other person as soon as they are entered, but similarly, I would like changes to binary files to be automatically reflected on the other editor's screen.
For example, if I edit LaTeX and the PDF is updated, I would like to perform an automatic reload when the file change is detected on the participant's side.

**Comment by msujew** [comment]
We send the correct updates in our file system provider. When taking a look at the extension host logging, it seems like the latex extension is attempting to reload the file from disk, which obviously doesn't work. It looks like the Latex extension cannot deal with the fact that the file comes from a virtual file system. It probably makes sense to log an issue with them.

**Comment by akenejie** [comment]
I am using an extension called “LaTeX Workshop” in my environment.
However, I submitted the issue because I would like to have the ability to reload binary files in general, including image files other than PDF.
Since the creation and deletion of files is reflected in Explorer in real time, I wonder if it would be possible to apply it

**Comment by msujew** [comment]
I use the same extension, see the error in the extension host log:

```
2024-09-01 01:39:16.790 [error] EntryNotFound (FileSystemError): Error: ENOENT: no such file or directory, stat 'C:\Workspace\Workspace\thesis.pdf'
    at Function.e (c:\Users\Mark\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:153:6206)
    at stat (c:\Users\Mark\AppData\Local\Programs\Microsoft VS Code\resources\app\out\vs\workbench\api\node\extensionHostProcess.js:153:3977)
    at Watcher.initiatePolling (c:\Users\Mark\.vscode\extensions\james-yu.latex-workshop-10.1.0\out\src\core\watcher.js:157:23)
    at Watcher.onDidChange (c:\Users\Mark\.vscode\extensions\james-yu.latex-workshop-10.1.0\out\src\core\watcher.js:128:13)
```

Please note the `C:\Workspace\Workspace\thesis.pdf` - even though the file is completely virtual, the extension attempts to load the file from disk. The correct URI would be `oct:///Workspace/Workspace/thesis.pdf`.

**Comment by msujew** [comment]
I've filed https://github.com/James-Yu/LaTeX-Workshop/issues/4357 to make the extension developers aware of this - there's nothing we can really do on our end.

**Comment by akenejie** [comment]
If LaTeX Workshop cannot do this, could it ignores the error and reopen the file with the same process as when the file was first opened for all binary files?

**Comment by msujew** [comment]
@akenejie I've filed https://github.com/James-Yu/LaTeX-Workshop/pull/4359 which fixes this on the LaTeX workshop extension side.

**Comment by akenejie** [comment]
Oh, too awesome!
I really appreciate that I will be able to write a joint Japanese paper, which I could not do with Live Share!

**Comment by msujew** [comment]
The PR is merged, I'll close this once the latest version of the latex workshop extension has been released.

**Comment by msujew** [comment]
My changes were included in the latest release of the Latex workshop extension. The change works as expected and file watching now works correctly. I'll close this.

**Comment by akenejie** [comment]
I have installed the latest version of LaTeX Workshop, but looking at the logs it does not appear that the “FILE_CHANGED” event was generated in the first place...
But then I changed the file name to alphabetical and now it is synchronized properly.

If it's “test.tex” it syncs, if it's “テスト.tex” the pdf doesn't sync and the cursor doesn't show up. Is this due to unicode characters?
Does this need a new issue?
